draft-ietf-quic-load-balancers.md

```diff
@@ -444,71 +444,157 @@ routing information of any given 4-tuple.
 
-Upon receipt of a QUIC packet, the load balancer extracts as many of the
-earliest octets from the destination connection ID as necessary to match the
-server ID. The nonce immediately follows.
+When generating a routable connection ID, the server writes arbitrary bits into
+its nonce octets, and its provided server ID into the server ID octets. See
+{{cid-entropy}} for nonce generation considerations.
 
-The load balancer decrypts the nonce and the server ID using the following three
-pass algorithm:
+If the server simply increments the nonce each time it has to generate a server
+ID, this simplifies avoidance of using the same nonce twice. However, in such a
+case the server SHOULD pick a random nonce from which to start counting from.
 
-* Pass 1:  The load balancer decrypts the server ID using 128-bit AES Electronic
-Codebook (ECB) mode, much like QUIC header protection. The encrypted nonce
-octets are zero-padded to 16 octets.  AES-ECB encrypts this encrypted nonce
-using its key to generate a mask which it applies to the encrypted server id.
-This provides an intermediate value of the server ID, referred to as server-id
-intermediate.
+The server encrypts the server ID using the following four pass algorithm, which
+leverages 128-bit AES Electronic Codebook (ECB) mode, much like QUIC header
+protection.
 
-server_id_intermediate = encrypted_server_id ^ AES-ECB(key, padded-encrypted-nonce)
+In the text below, ^ is the XOR function and || is concatenation. The truncate()
+function takes the most significant octets of its argument, so that the XOR
+function operates on fields of the same length. The expand() function outputs
+16 octets, with its first argument in the most significant bits, its second
+argument in the least significant bits, and zeros in all other positions. Thus,
 
-* Pass 2:  The load balancer decrypts the nonce octets using 128-bit AES
-ECB mode, using the server-id intermediate as "nonce" for this pass. The
-server-id intermediate octets are zero-padded to 16 octets.  AES-ECB encrypts
-this padded server-id intermediate using its key to generate a mask which it
-applies to the encrypted nonce. This provides the decrypted nonce value.
+~~~pseudocode
+expand(0xaaba3c, 0x13) = 0xaaba3c00000000000000000000000013
+~~~
 
-nonce = encrypted_nonce ^ AES-ECB(key, padded-server_id_intermediate)
+The example at the end of this section helps to clarify the steps described
+below.
 
-* Pass 3:  The load balancer decrypts the server ID using 128-bit AES ECB mode.
-The nonce octets are zero-padded to 16 octets.  AES-ECB encrypts this nonce
-using its key to generate a mask which it applies to the intermediate server id.
-This provides the decrypted server ID.
+1. The server concatenates the server ID and nonce to create plaintext_CID.
 
-server_id = server_id_intermediate ^ AES-ECB(key, padded-nonce)
+2. The server splits plaintext_CID into components left_0 and right_0 of equal
+length, splitting an odd octet in half if necessary.
 
```

Comment:
That's probably where I missed the point in my first review. An example would be nice.